lib/utapi/utilities.js

```diff
@@ -0,0 +1,117 @@
+import { auth } from 'arsenal';
+import commander from 'commander';
+import http from 'http';
+import https from 'https';
+import { logger } from '../utilities/logger';
+
+function _listBucketMetrics(host,
+                            port,
+                            buckets,
+                            timeRange,
+                            accessKey,
+                            secretKey,
+                            verbose,
+                            ssl) {
+    const options = {
+        host,
+        port,
+        method: 'POST',
+        path: '/buckets?Action=ListMetrics',
+        headers: {
+            'content-type': 'application/json',
+            'cache-control': 'no-cache',
+        },
+        rejectUnauthorized: false,
+    };
+    const transport = ssl ? https : http;
+    const request = transport.request(options, response => {
+        if (verbose) {
+            logger.info('response status code', {
+                statusCode: response.statusCode,
+            });
+            logger.info('response headers', { headers: response.headers });
+        }
+        const body = [];
+        response.setEncoding('utf8');
+        response.on('data', chunk => body.push(chunk));
+        response.on('end', () => {
+            const responseBody = JSON.parse(body.join(''));
+            if (response.statusCode >= 200 && response.statusCode < 300) {
+                // eslint-disable-next-line no-console
+                console.log(responseBody);
+                process.exit(0);
+            } else {
+                logger.error('request failed with HTTP Status ', {
+                    statusCode: response.statusCode,
+                    body: responseBody,
+                });
+                process.exit(1);
+            }
+        });
+    });
+    auth.generateV4Headers(request, '', accessKey, secretKey, 's3');
+    if (verbose) {
+        logger.info('request headers', { headers: request._headers });
+    }
+    request.write(JSON.stringify({ buckets, timeRange }));
+    request.end();
+}
+
+/**
+ * This function is used as a binary to send a request to utapi server
+ * to list bucket metrics
+ *
+ * @return {undefined}
+ */
+export function listBucketMetrics() {
+    commander
```

Comment:
I just meant to have all the commander-related stuff in the bin/ script rather than in an utilities.js
I personally do not think that this kind of argument management code has its place in a set of utilities, but I might just be oldschool.